tests/gladeTest.c

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file	main.c
+ * @date	Apr 20, 2016
+ * @author	beattie
+ *
+ * @brief	file contents
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <gtk/gtk.h>
+#include <goat-plot.h>
+#include <math.h>
+
+typedef struct {
+	GoatPlot *plot;
+	GoatDataset *dataset;
+} Both;
+
+gboolean dynamic_add(Both *both)
+{
+	static uint16_t idx = 0;
+	double x = 0;
+	double y = 0;
+
+	if(idx > 100) {
+		return 0;
+	}
+	g_print("whatsoever callback\n");
+	idx++;
+	x = idx * 0.1;
+	y = cos(x + 0.5 * M_PI) * (sqrt(0.01 * idx * idx * idx));
+	goat_dataset_append(both->dataset, x, y);
+	gtk_widget_queue_draw(GTK_WIDGET(both->plot));
+	return 1;
+}
+
+int main(int argc, char *argv[])
+{
+	GtkBuilder *builder;
+	GtkWidget *window;
+	GoatPlot *plot;
+	GoatDataset *dataset;
+	Both both;
+
+	gtk_init(&argc, &argv);
+
+	plot = goat_plot_new();
+
+	if((builder = gtk_builder_new_from_file("test.glade")) ==
+			NULL) {
+		fprintf(stderr, "gtk_builder_new failed\n");
+		exit(-1);
```

Comment:
https://developer.gnome.org/glib/stable/glib-Message-Logging.html#g-error
`
g_error(...)` is a shorter and cleaner way to do so